include/biosHandler.hpp

```diff
@@ -4,16 +4,81 @@
 
 namespace vpd
 {
+
+/**
+ * @brief Interface class for BIOS handling.
+ *
+ * The class layouts virtual methods required to be implemented by any concrete
+ * class intended to use feature provided via BIOS handler class.
+ */
+class BiosHandlerInterface
+{
+  public:
+    /**
+     * @brief API to back up or restore BIOS attributes.
+     *
+     * Concrete class should implement the API and read the backed up data from
+     * its designated location and take a call if it should be backed up or
+     * restored.
+     */
+    virtual void backUpOrRestoreBiosAttributes() = 0;
+
+    /**
+     * @brief API to be triggered on BIOS attribute change.
+     *
+     * Concrete class should implement the API to capture any change in the BIOS
+     * attribues and in the implementation needs to filter out the attribute it
+     * is interested.
+     *
+     * @param[in] i_msg - The callback message.
+     */
+    virtual void biosAttribsCallback(sdbusplus::message_t& i_msg) = 0;
+};
+
+/**
+ * @brief IBM specifc BIOS handler class.
+ */
+class IbmBiosHandler : public BiosHandlerInterface
+{
+  public:
+    /**
+     * @brief API to back up or restore BIOS attributes.
+     *
+     * The API will read the backed up data from the VPD keyword and based on
+     * its value, either backs up or restores the data.
+     */
+    virtual void backUpOrRestoreBiosAttributes();
+
+    /**
+     * @brief API to be triggered on BIOS attribute change.
+     *
+     * The API listens for specific BIOS attributes and backs them up to VPD
```

Comment:
is this listener API ?
same comment I gave in base class.

Reply:
Will update.